### DISCUSSION TITLE
Select word with double click and line with 3 clicks

### DISCUSSION THREAD

**Discussion by darrigu** [helix-editor/helix#10208] [Ideas]
# Idea

All is in title

**Comment by wisnoskij** [comment]
Came to suggest this. Would be great to have leftclick, etc button support. And I would argue all day long for OP idea to be the sane defaults. I dont think anyone what wants any sort of mouse support at all not wanting double/triple click selection support.

**Comment by theaog** [comment]
why re-implement mouse support in the editor when you can use your terminal app support?
I use Alacritty and just press Shift and get all mouse selection features; double/triple-click, boxed selection, etc.

**Comment by ngasull** [comment]
Because selecting from the displayed content isn't the same as creating an helix selection in the *actual* content.

Would love having double click select surrounding word for quick replacements!

**Comment by theaog** [comment]
> Because selecting from the displayed content isn't the same as creating an helix selection in the _actual_ content.
> 
> Would love having double click select surrounding word for quick replacements!

miw / maw

**Comment by KiaraGrouwstra** [comment]
would love to see this resolved as well. the current behavior seems not to be quite as useful.

**Comment by peterjc** [comment]
I'd like this too as the default - initially misfiled as a duplicate issue: https://github.com/helix-editor/helix/issues/12500

I'd even like to be able to go further, building on https://docs.helix-editor.com/textobjects.html and LSP support - for instance being able to configure extra mouse buttons as select function or class. I'd like to try ctrl+click or meta+click, or things like three-figure click on a supported device.

(My thinking is currently I _read_ a lot of code under mouse control - wheel scrolling etc - and then occasionally need to move to the keyboard)

**Comment by nik-rev** [comment]
Extended mouse support would be really nice to have. I've gone ahead and implemented this in the PR: https://github.com/helix-editor/helix/pull/12514

**Comment by soomtong** [comment]
Does use a mouse in HX? I never thought of that!
